### PR TITLE
TSA-351 JSON logging

### PR DESCRIPTION
**Pull request by robyoung-digital**
## Add JSON logging with request IDs

Logs are output as JSON from Celery and Flask.
Request ID is added by Flask and forwarded to Celery tasks.
All logs include the following records if they can:
name, levelname, pathname, lineno, service, log_type, request_id

## add request-id requests wrapper

Add a module that wraps requests and adds the TS-Request-ID header to
each call. This ensures request IDs are propagated across service calls.

### CODE REVIEW

File: thunderstorm_auth/logging/__init__.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+
+_ID_GETTERS = []
+
+
+def _register_id_getter(getter):
+    _ID_GETTERS.append(getter)
+
+
+def get_request_id():
```
Comment (alexanderson) [inline thunderstorm_auth/logging/__init__.py]: docs?

File: thunderstorm_auth/logging/celery.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            record.task_name = task.name
+        else:
+            record.task_id = '???'
+            record.request_id = None
+            record.task_name = '???'
+
+        return record
+
+
+class CeleryRequestIDTask(CeleryTask):
```
Comment (alexanderson) [inline thunderstorm_auth/logging/celery.py]: How is this class used? How can we enforce all tasks in all TS services to use this?

File: thunderstorm_auth/logging/celery.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    def filter(self, record):
+        task = get_current_task()
+        if task and task.request:
+            record.task_id = task.request.id
+            record.request_id = get_celery_request_id(task.request)
+            record.task_name = task.name
+        else:
+            record.task_id = '???'
+            record.request_id = None
+            record.task_name = '???'
```
Comment (alexanderson) [inline thunderstorm_auth/logging/celery.py]: Why `'???'` and not `None`?

Comment (robyoung-digital) [reply thunderstorm_auth/logging/celery.py]: This was lifted from celery's logger so I left it to stay as close the original as possible. I'm happy to change it.

File: thunderstorm_auth/logging/celery.py (hunk 2 above)
Comment (robyoung-digital) [reply thunderstorm_auth/logging/celery.py]: I'll add some docs to the PR for this but just to outline.

1. It replaces the `Task` attribute of the celery app and will then be used for all tasks sent from that app.
2. We can't really. We add it to the celery apps we know about and then fill in the gaps.

**Pull request merged**